src/main/java/uk/nhs/digital/emailadapter/schedulingtasks/EmailScheduledTask.java

```diff
@@ -0,0 +1,178 @@
+@Component
+public class EmailScheduledTask {
+
+  private final DateTimeFormatter FOMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd_HH:mm:ss");
+  private final ExchangeService service = new ExchangeService(ExchangeVersion.Exchange2010_SP2);
+  private final AWSSimpleSystemsManagement ssm =
+      AWSSimpleSystemsManagementClientBuilder.defaultClient();
+  private final StandardPBEStringEncryptor decryptor = new StandardPBEStringEncryptor();
+  private final String mpCryptoPassword;
+
+  @Autowired
+  public EmailScheduledTask(@Value(value = "${mpCryptoPassword}") final String mpCryptoPassword)
+      throws Exception {
+    this.mpCryptoPassword = mpCryptoPassword;
+    decryptor.setPassword(mpCryptoPassword);
+    ExchangeCredentials credentials = new WebCredentials(
+        decryptor.decrypt(getParameter("username")), decryptor.decrypt(getParameter("password")));
+    service.setCredentials(credentials);
+    service.autodiscoverUrl(decryptor.decrypt(getParameter("username")));
+  }
+
+  @Async
+  @Scheduled(fixedRateString = "${fixedRate.in.milliseconds}",
+      initialDelayString = "${initialDelay.in.milliseconds}")
+  public void sendMails() throws InterruptedException {
+    String timeStamp = FOMATTER.format(LocalDateTime.now());
+    log.info("Invocation started: {}", timeStamp);
+
+    try {
+      StagedStopwatch stopwatch = StagedStopwatch.start();
+      ItemView view = new ItemView(Integer.parseInt(getParameter("EMAIL_ITEM_VIEW")));
+      SearchFilterCollection searchFilterCollection =
+          new SearchFilter.SearchFilterCollection(LogicalOperator.And);
+      searchFilterCollection.add(
+          new SearchFilter.IsEqualTo(EmailMessageSchema.From, getParameter("EMS_REPORT_SENDER")));
+      searchFilterCollection.add(new SearchFilter.IsEqualTo(EmailMessageSchema.IsRead, false));
+      FindItemsResults<Item> findResults =
+          service.findItems(WellKnownFolderName.Inbox, searchFilterCollection, view);
+
+      while (findResults.getTotalCount() > 0) {
+        for (Object item : findResults.getItems()) {
+          EmailMessage emailMessage = (EmailMessage) item;
+          emailMessage
+              .load(new PropertySet(BasePropertySet.FirstClassProperties, ItemSchema.MimeContent));
+          log.info("attachment count: {} ", emailMessage.getAttachments().getCount());
+
+          Attachment attachment = emailMessage.getAttachments().getItems().get(0);
+          stopwatch.finishStage("Getting html attchement from email");
+
+          if (attachment instanceof FileAttachment) {
+            FileAttachment fileAttachment = (FileAttachment) attachment;
+            fileAttachment.load();
+
+            if (fileAttachment.getContentType().equalsIgnoreCase("text/html")) {
+              // convert bytes[] to string
+              String htmlString = new String(fileAttachment.getContent(), StandardCharsets.UTF_8);
+              byte[] transform = new PDFTransformer().transform(Jsoup.parse(htmlString).html());
```

Comment:
Jsoup.parse(htmlString) : line 123 & 131 does the parsing twice. Do it once and use it at both places.